src/settings/controllers/PseudonymousAnalyticsController.ts

```diff
@@ -0,0 +1,26 @@
+/*
+Copyright 2021 The Matrix.org Foundation C.I.C.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+import SettingController from "./SettingController";
+import { SettingLevel } from "../SettingLevel";
+import { getAnalytics } from "../../PosthogAnalytics";
+import { MatrixClientPeg } from "../../MatrixClientPeg";
+
+export default class PseudonymousAnalyticsController extends SettingController {
+    public onChange(level: SettingLevel, roomId: string, newValue: any) {
+        getAnalytics().updateAnonymityFromSettings(MatrixClientPeg.get().getUserId());
+    }
```

Comment:
Kinda feels like this could/should just be the Analytics instance subscribing to changes on the SettingsStore rather than abusing a controller

Reply:
The documentation for `SettingsStore.watchSetting` suggests "Watching is intended to be used in scenarios where the app needs to react to changes made by other devices. It is otherwise expected that callers will be able to use the
Controller system or track their own changes to settings."

It's not possible to listen to UI events for an individual labs flag in the UI due to the way the screen is structured.

What makes it abuse? What are controllers for?

Reply:
Normally changing the default behaviour of the SettingsStore but fair enough in this case. I could forsee that we might want it to be synced between devices in the future so a watch/monitor might be more flexible